src/libsyntax/print/pprust.rs

```diff
@@ -1725,7 +1725,10 @@ impl<'a> State<'a> {
         self.print_expr_cond_paren(
             scrutinee,
             Self::cond_needs_par(scrutinee)
-            || parser::needs_par_as_let_scrutinee(scrutinee.precedence().order())
+                || parser::needs_par_as_let_scrutinee(scrutinee.precedence().order())
+                // #51635: handle closures correctly, even though they have a really low
+                // precedence, they *don't* need parens for assignment 🤷.
```

Comment:
Heh; would be good to replace 🤷 with the actual reason if you have a sec to figure it out.

Reply:
I'm waiting on insight from @petrochenkov, I believe the precedence order between `let` and closures is incorrectly, but I didn't want to affect all other relative precedences accidentally so I did this, but the underlying reason would require going over every `Precedence` and double checking them. Clearly the `Parser` disagrees with what we have encoded :-|

Reply:
The alternatives would of course be to either change the parser to conform to the encoded precedence (potentially breaking working code) or changing the precedence order (potentially breaking working code) :-|